File: 闲来无事/调用信达App/调用信达App/ViewController.m
//
//  ViewController.m
//  调用信达App
//
//  Created by huangaengoln on 15/12/31.
//  Copyright © 2015年 huangaengoln. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self createButtonWithFrame:CGRectMake(10, 80, self.view.frame.size.width-20, 40) tag:1 title:@"点击调用信达BD"];
    [self createButtonWithFrame:CGRectMake(10, 130,  self.view.frame.size.width-20, 40) tag:2 title:@"点击调用信达GD"];
}

-(void)createButtonWithFrame:(CGRect)frame tag:(NSInteger)tag title:(NSString *)title {
    UIButton *button=[UIButton buttonWithType:UIButtonTypeCustom];
    button.frame=frame;
    [button setTitle:title forState:UIControlStateNormal];
    [button setBackgroundColor:[UIColor blueColor]];
    button.tag=tag;
    [button addTarget:self action:@selector(clickToApp:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:button];
}


-(void)clickToApp:(UIButton *)button {
    
#if 0
    NSURL *url  = [NSURL URLWithString:@"AppTwo"];
    
    if ([[UIApplication sharedApplication] canOpenURL:url])
    {
        NSLog(@"canOpenURL");
        [[UIApplication sharedApplication] openURL:url];
    } else
    {
        NSLog(@"can not OpenURL");
    }
#endif
    NSURL *nsurl1=[NSURL new];
    NSURL *nsUrl=[NSURL new];
    if (button.tag == 1) {
        nsUrl=[NSURL URLWithString:@"comXindaThinkiveXinda://?Identifier=com.thinkiveBD"];
    } else if (button.tag == 2) {
        nsUrl=[NSURL URLWithString:@"comXindaThinkiveXindaGD://?Identifier=com.thinkiveGD"];
    }
//    NSLog(@"nsUrl : %@,nsurl1 : %@",nsUrl,nsurl1);
//    BOOL canOpenURL=[[UIApplication sharedApplication]canOpenURL:[NSURL URLWithString:@"comXindaThinkiveXinda://?Identifier=com.thinkiveBD"]];

//    if ([[UIApplication sharedApplication]canOpenURL:[NSURL URLWithString:@"comXindaThinkiveXinda://?Identifier=com.thinkiveBD"]]) {
//    }
    if (nsUrl) {
        [[UIApplication
          sharedApplication] openURL:nsUrl];
    }
    
    //   comXindaThinkiveXinda://?Identifier=com.thinkiveBD
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 闲来无事/swift(iOS 9 Xcode7)   一个app打开另一个app内部的页面/Test/Test/ViewController.swift
//
//  ViewController.swift
//  Test
//
//  Created by huangaengoln on 15/10/29.
//  Copyright © 2015年 huangaengoln. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        openAppButton()
    }
    func openAppButton() {
        let button=UIButton()
        self.view.addSubview(button)////(必须要先把button加进来，才可以用去写它的布局)erminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'couldn't find a common superview for <UIButton: 0x136682f20; frame = (0 0; 0 0); opaque = NO; layer = <CALayer: 0x136682e70>> and <UIView: 0x136683900; frame = (0 0; 414 736); autoresize = W+H; layer = <CALayer: 0x136682be0>>'*** First throw call stack:
        button.frame=CGRectMake(20, 100, self.view.frame.size.width-2*20, 40);
//        button.center=self.view.center
        button.setTitle("打开BasicGrammar", forState: UIControlState.Normal)
        button.setTitleColor(UIColor.cyanColor(), forState: UIControlState.Normal)
        button.setTitleColor(UIColor.greenColor(), forState: UIControlState.Highlighted)
        button.titleLabel?.font=UIFont.systemFontOfSize(18)
        
        //设置button边框
        button.layer.borderColor=UIColor.greenColor().CGColor
        button.layer.borderWidth=2
        button.layer.cornerRadius=10
//        button.layer.masksToBounds=true
        button.addTarget(self, action: "openApp", forControlEvents: UIControlEvents.TouchUpInside)
    }
    func openApp() {
        //记得设置info.plist里面的LSApplicationQueriesSchemes，iOS9止呕需要，iOS9之后提高了app的安全性，需要给出一个类似白名单的东西，在白名单里面的才能打开app。不然报错：： -canOpenURL: failed for URL: "OpenAppTest://mark?id=007" - error: "This app is not allowed to query for scheme OpenAppTest"
        //OpenAppTest://mark?id=xxxx   (调用BasicGrammar app 拼接参数字符串，拼接的时候就像url那样子  OpenAppTest://标记名字?name＝xiaomin&age＝23)
        let urlStr="OpenAppTest://mark?id="+"007"
        let customUrl=NSURL(string: urlStr)
        if UIApplication.sharedApplication().canOpenURL(customUrl!) {
            UIApplication.sharedApplication().openURL(customUrl!)
        } else {
            //提示没有安装BasicGrammar app
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

